libopendavinci/testsuites/ContainerTestSuite.h

```diff
@@ -143,6 +162,79 @@ class ContainerTest : public CxxTest::TestSuite {
             TS_ASSERT(c2.getReceivedTimeStamp().toString() == c1.getReceivedTimeStamp().toString());
             TS_ASSERT(c2.getSampleTimeStamp().toString() == c1.getSampleTimeStamp().toString());
         }
+
+        void testContainerListenerFork() {
+            ContainerListenerFork clf;
+
+            MyContainerListener listener1;
+            MyContainerListener listener2;
+            MyContainerListener listener3;
+
+            TS_ASSERT(!listener1.m_hasReceived);
+            TS_ASSERT(!listener2.m_hasReceived);
+
+            clf.addContainerListener(&listener1);
+            TS_ASSERT(!listener1.m_hasReceived);
+            TS_ASSERT(!listener3.m_hasReceived);
+
+            clf.addContainerListener(&listener2);
+            TS_ASSERT(!listener1.m_hasReceived);
+            TS_ASSERT(!listener2.m_hasReceived);
+            TS_ASSERT(!listener3.m_hasReceived);
```

Comment:
What is the reason for asserting listeners in this order: first listener 1 and 2, then 1 and 3, and then 1-3?